src/lib/tls/tls_signature_scheme.h

```diff
@@ -0,0 +1,106 @@
+/*
+* TLS Signature Scheme
+* (C) 2022 Jack Lloyd
+*     2022 Hannes Rantzsch, René Meusel - neXenio GmbH
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_TLS_SIGNATURE_SCHEME_H_
+#define BOTAN_TLS_SIGNATURE_SCHEME_H_
+
+#include <botan/types.h>
+#include <botan/asn1_obj.h>
+#include <botan/pk_keys.h>
+
+#include <optional>
+#include <string>
+
+namespace Botan::TLS {
+
+class Protocol_Version;
+
+class BOTAN_PUBLIC_API(3,0) Signature_Scheme
+{
+public:
+/**
+ * Matches with wire encoding
+ *
+ * Note that this is intentionally left as a bare enum. It emulates the Botan 2
+ * API where `Signature_Scheme` was an enum class with associated free-standing
+ * functions. Leaving it as a bare enum resembles the legacy user-facing API.
+ */
+enum Code : uint16_t {
```

Comment:
@randombit [said](https://github.com/randombit/botan/pull/2958#discussion_r851222378):

> My initial reaction was "can we make this an `enum class`"? But I see that this breaks other code that is using `Signature_Scheme` as if it was still an `enum`. So maybe add a comment here, that the bare `enum` is intentional because the overall class is emulating an enum. 

From your comment I figure you're fine with this end user-facing API change. I'll add a comment describing the intention regarding the bare enum and that it is supposed to resemble the old `Signature_Scheme` enum class.